Fix: Add status in group member modal

Boruus opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
- Add `StatusIcon` component in `GroupMemberList` and `ViewUserGroupModal` components.
- We get status by using `getStatusForUserId` selector.
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->

#### Ticket Link
No ticket or issue are link to this change.
<!--
If applicable, please include both or either of the following links:

Fixes https://github.com/mattermost/mattermost/issues/XXX
Jira https://mattermost.atlassian.net/browse/MM-XXX
-->

#### Screenshots
(I had a little trouble testing the fix in a local environment because the feature `group member` is only accessible from an enterprise offer, but it looks something like this: )
|  before  |  after  |
|----|----|
| ![image](https://github.com/user-attachments/assets/6c7428aa-228c-46ba-99ed-8ed8162238a1)| ![image](https://github.com/user-attachments/assets/d7d3db94-fb4e-4d94-9948-bae682c62be0) |
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
Fix: Add status in group member modal
```

mattermost-build commented:
Hello @Boruus,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**
Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.

This is a standard procedure for many open source projects.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

pvev commented:
hello @Boruus , please sign the [contributor license agreement](https://mattermost.com/mattermost-contributor-agreement/) so we can proceed with the review and merge of this PR.

Boruus commented:
> hello @Boruus , please sign the [contributor license agreement](https://mattermost.com/mattermost-contributor-agreement/) so we can proceed with the review and merge of this PR.
![Capture d’écran 2024-07-30 à 12 04 07](https://github.com/user-attachments/assets/08b3366f-3cd0-4a56-b0e4-d8c7beeac2ca)

/check cla

pvev commented on webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:
```diff
@@ -149,6 +155,8 @@ const GroupMemberList = (props: Props) => {
     };
 
     const Item = ({index, style}: ListChildComponentProps) => {
+        const status = useSelector((state: GlobalState) => getStatusForUserId(state, members[index].user.id) || UserStatuses.OFFLINE);
```
```suggestion
        const status = useSelector((state: GlobalState) => getStatusForUserId(state, members[index]?.user?.id) || UserStatuses.OFFLINE);
```

I pull your changes and it was breaking when clicking to see the group information. I identified this line as the problem, with this optional values the issue is solved 
<img width="639" alt="Screenshot 2024-08-07 at 23 58 51" src="https://github.com/user-attachments/assets/26f18b0e-e5ed-4041-8900-409780967396">

pvev reviewed (changes requested):
thanks for your contribution @Boruus 🚀 , so far is a great work, there is only an issue happening. Please address my comments and let me know if you need any assistance. Thanks

Boruus commented:
> thanks for your contribution @Boruus 🚀 , so far is a great work, there is only an issue happening. Please address my comments and let me know if you need any assistance. Thanks

Thanks for your feedback, I've added your changes which fix the problem nicely, I've also adjusted the button's css so that the overlay doesn't overlap the other users. Don't hesitate if you have any other feedback. 
![Capture d’écran 2024-08-08 à 09 23 28](https://github.com/user-attachments/assets/78ccc49e-3159-4941-9e50-eb10ad8b8546)

larkox commented on webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:
```diff
@@ -171,13 +179,20 @@ const GroupMemberList = (props: Props) => {
+                                    tabIndex={-1}
+                                />
+                                <StatusIcon
+                                    className='status user-popover-status'
```
Maybe I am missing something, but the only place `user-popover-status` is defined, is in `webapp/channels/src/components/profile_popover/profile_popover.scss`, and it is defined under the `user-popover-image` class, so... it should not be applying any stile here, right? If so, why the class name?

Boruus replied:
@larkox you're right the props classname and button are useless in these cases, I thought I had to add a style for the status icon but there's already a `.status` included in `.status-wrapper` class so no need to add it, thanks for the feedback

larkox commented on webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:
```diff
@@ -171,13 +179,20 @@ const GroupMemberList = (props: Props) => {
+                                <StatusIcon
+                                    className='status user-popover-status'
+                                    status={status}
+                                    button={true}
```
The `button` property seems to only remove the `status` class, and we are adding it again in the class name. Therefore... shouldn't it work the same way omitting the `button` prop and the `status` part of the classname?

larkox commented on webapp/channels/src/components/view_user_group_modal/view_user_group_list_item/view_user_group_list_item.tsx:
```diff
@@ -41,19 +47,27 @@ const ViewUserGroupListItem = (props: Props) => {
                 />
-            </>
+                <StatusIcon
+                    className='status bottom user-popover-status'
```
Same as my other two comments.

larkox reviewed (changes requested):
Looks great, but a couple of things to verify.

Also, almost sure there are still some lint issues, so please run `npm run fix` .

Boruus commented:
snapshot are failing i'm fixing it

Boruus commented:
> snapshot are failing i'm fixing it

fixed

pvev reviewed (approved):
Thanks for addressing our comments. This looks great now 🎉, Approved ✅  . Thank you for this fantastic work and thank you for contributing to Mattermost 🚀

larkox commented:
/update-branch

yasserfaraazkhan reviewed (approved):
<img width="314" alt="image" src="https://github.com/user-attachments/assets/ff2c733b-beea-40cb-99fa-80db63c78e72">
